src/lexer.coffee

```diff
@@ -667,6 +667,9 @@ exports.Lexer = class Lexer
         @error message, origin[2] if message
       return value.length if skipToken
 
+    if prev?[0] is 'IMPORT' and value is '('
+      prev[0] = 'IDENTIFIER'
```

Comment:
This is technically cheating; the `import` here is a special keyword like `super`, so it really should be its own token type. Should we make a new token like `DYNAMIC_IMPORT` or something?

Reply:
Ah ok, I was just copying my approach (for `new.target`) from #5106. But I think you're right that (in both cases) it's cleaner to specify new token types/grammar constructs

So updated to use new `DYNAMIC_IMPORT` token and `DynamicImport`/`DynamicImportCall` node classes

This should set it up to be easy to generate the appropriate AST for dynamic imports (which looks like a regular call AST with a "variable"/`callee` of `type: 'Import'`)

As far as `new.target`, I checked Babel AST and it actually should be generating a special AST node for `new.target` (of `type: 'MetaProperty'`). So I think that's another argument that `new.target` should similarly be updated to use a special token type/node class. So for `new.target` I'd plan on updating it in a PR against `ast` unless you'd rather I separate out a PR against `master` for the token type/node class and then a second PR against `ast` for just the `MetaProperty` AST generation